pkg/ingress/controller.go

```diff
@@ -423,3 +455,21 @@ func (c *Controller) syncSSL(ctx context.Context, ssl *apisixv1.Ssl, event types
 	}
 	return err
 }
+
+func (c *Controller) checkClusterHealth(ctx context.Context, cancelFunc context.CancelFunc) {
+	for {
+		select {
+		case <-ctx.Done():
+		case <-time.After(5 * time.Second):
+		}
+
+		err := c.apisix.Cluster(c.cfg.APISIX.DefaultClusterName).HealthCheck(ctx)
+		if err != nil {
+			// Finally failed health check, then give up leader.
+			log.Warnf("failed to check health for default cluster: %s, give up leader", err)
+			cancelFunc()
+			return
```

Comment:
Should we set the state to `_cacheSyncing` before return?

Reply:
No, the current logic is to just discard it if it is unhealthy.